### PR TITLE
inf-terraform-aws:  switch from shared statefile location to dedicated

### LINKED ISSUE #900: inf-terraform-aws: switch from shared statfile location to dedicated
**Is your feature request related to a problem? Please describe.**
Currently the Terraform state file is stored in an shared s3 bucket managed via policies which cumbersome and error prone. 

**Describe the solution you'd like**
Each account has its own dedicated s3 bucket. This one will be then re

**Additional context**
Moving to an s3 bucket located in the same account is easier to manage and requires less configuration of the quickstarter.

### PR DESCRIPTION
**Pull request by nichtraunzer**
This PR enables the AWS quickstarter to store the terraform statefile inside its own account. Processing takes place in the Makefile so no change in the shared library necessary.

Closes #900 

Tasks: 
- [ ] Updated documentation in `docs/modules/...` directory
- [x] Ran tests in `<quickstarter>/testdata` directory

### CODE REVIEW

File: inf-terraform-aws/files/.pre-commit-config.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ repos:
   - id: terraform_fmt
 
 - repo: https://github.com/nichtraunzer/ods-pre-commit-hooks
-  rev: v0.4.1
+  rev: v0.4.2
```
Comment (tbugfinder) [inline inf-terraform-aws/files/.pre-commit-config.yaml:53]: Please add this also for azure.

File: inf-terraform-aws/files/Makefile [hunk 2]
```diff
@@ -55,6 +64,8 @@ test: install-test-deps
 		./.venv/bin/python3 ./.venv/bin/hcl2tojson test/fixtures/$$fdir/main.tf  test/integration/$$fdir/files/main.json; \
 	done \
 
+	exit 0
```
Comment (tbugfinder) [inline inf-terraform-aws/files/Makefile:67]: hm either this is a left over or the following lines are no longer needed.

File: inf-terraform-aws/files/Makefile [hunk 3]
```diff
@@ -140,7 +156,7 @@ install-test-deps: install-ruby-gems install-python-env
 # run cinc-auditor without use of kitchen-terraform and create yaml for mapping terraform outputs to inspec inputs.
 cinc-auditor-test:
 	sh ./lib/scripts/createstackfixtureoutputs2yml.sh
-	bundle exec cinc-auditor exec test/integration/default --no-create-lockfile --no-distinct-exit --input-file ./test/integration/default/files/inputs-from-tfo-stack.yml --target aws://
+	bundle exec cinc-auditor exec test/integration/default --no-create-lockfile --no-distinct-exit --input-file ./test/integration/$$profile/files/inputs-from-tfo-stack.yml --target aws://
```
Comment (tbugfinder) [inline inf-terraform-aws/files/Makefile:159]: $$profile is not correct.

File: inf-terraform-aws/files/test/fixtures/default/random.tf [hunk 4]
```diff
@@ -4,16 +4,3 @@ resource "random_id" "id" {
...
-  length           = 16
-  min_lower        = 1
-  min_upper        = 1
-  number           = true
-  special          = true
-  override_special = "!#+%"
-}
-
-locals {
-  id       = random_id.id.hex
```
Comment (tbugfinder) [inline inf-terraform-aws/files/test/fixtures/default/random.tf:17]: local.id should be available

**Review by tbugfinder — CHANGES_REQUESTED** [review]

File: inf-terraform-aws/files/Makefile (hunk 3 above)
Comment (nichtraunzer) [reply inf-terraform-aws/files/Makefile]: how this happened?

File: inf-terraform-aws/files/test/fixtures/default/random.tf (hunk 4 above)
Comment (nichtraunzer) [reply inf-terraform-aws/files/test/fixtures/default/random.tf:17]: :facepalm

**Review by tbugfinder — APPROVED** [review]

**Pull request merged**